# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/misc/joystick/hotfixes.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- joystick-1.2.14/Makefile	1999-03-05 20:37:56.000000000 +0100
+++ joystick-1.2.14/Makefile	2005-03-04 10:10:03.510668744 +0100
@@ -28,6 +28,7 @@
 CC		= gcc
 
 CPPFLAGS	= -I.#						# for joystick.h
+CPPFLAGS	+= -I /usr/include/linux24
 
 CFLAGS		= -O2 -Wall -pipe -m486 -fomit-frame-pointer#	# i386
 # CFLAGS	= -O2 -Wall -pipe -m68020 -ffixed-a2#		# m68k
@@ -60,16 +61,13 @@
 JOYOPRE		:= $(shell if [ -f .prerelease ]; then cat .prerelease; else echo 0; fi)
 JOYPRE		:= $(shell expr $(JOYOPRE) + 1)
 
-compile: joystick.o $(DRIVERS) $(PROGRAMS)
+compile: $(PROGRAMS)
 
 install: compile
-	mkdir -p $(MODDIR)
-	rm -f $(MODDIR)/joy*.o
-	cp joy*.o $(MODDIR)
-	/sbin/depmod -a
-	install -m 644 joystick.h /usr/include/linux
-	install -m 755 jscal /usr/local/bin
-	install -m 755 jstest /usr/local/bin
+	install -m 755 jscal /usr/bin
+	install -m 755 jstest /usr/bin
+	install -m 755 jsattach /usr/bin
+	cp *.txt /usr/share/doc/joystick/
 
 joystick.o: joystick.c joystick.h
 	$(CC) $(CFLAGS) $(CPPFLAGS) $(MODFLAGS) -c joystick.c -o joystick.o
